lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4SharedOnChipRam.scala

```diff
@@ -60,60 +60,123 @@ case class Axi4SharedOnChipRam(dataWidth : Int, byteCount : BigInt, idWidth : In
 }
 
 
-//WARNING, do not support backpresure
-case class Axi4SharedOnChipRamMultiPort(portCount : Int, dataWidth : Int,byteCount : BigInt,idWidth : Int) extends Component{
-  val axiConfig = Axi4SharedOnChipRam.getAxiConfig(dataWidth,byteCount,idWidth)
+case class Axi4SharedOnChipRamPort(config: Axi4Config) extends Bundle {
```

Comment:
Why extends Bundle ?, i mean that class has only functions right ?

Reply:
IMO, the port is a bundle of wires which have specific functions.

Reply:
the Axi4SharedOnChipRamPort class is a collection of function which use the same config i would say, so, i would say, they can be moved into a scala object, with the added config parameter as arguement (trying to avoid this pattern where you have to create a class just to hide passing one parameter later on)

Reply:
do you mean use an object pattern, as below
```
object Axi4SharedOnChipRamPort{
   def apply(config: Axi4Config):
        .....
}

class Axi4SharedOnChipRamPort extends xxxx{
   ..
}
```
The problem is that if I extends a Component class the io is required to connect ram, which seems make it too complicated.

Reply:
no, i mean : 

```scala
object Axi4SharedOnChipRamPort{
   def attach(config: Axi4Config, ram: Mem[Bits]): Axi4Shared = { ... }
}
```

Reply:
I hide the attach function with a direct object initialization.